### DISCUSSION TITLE
Filtering by UUID field Within Data Provider

### DISCUSSION THREAD

**Discussion by wesshaw** [Budibase/budibase#4592] [Help] [answered]
Hello all.  I'm in a situation where I need to be able to filter based on a field other than a simple integer id.  I need to filter based on a UUID field.  It _seems_ to function as intended because I get the data returned as expected, but I get a 400 and error on the search that reads "error: invalid input syntax for type uuid: "359131"".  It would appear that the search api is truncating the uuid and only taking the first 6 characters.  Do I have something misconfigured?

My DB is an Aurora PostgreSQL

**Comment by mjashanks** [comment, marked as answer]
Hey Wes - I just tried this, using a local postgres DB, and worked for me. My steps
1. Added a column of to a table, of type `uuid`
2. In Budibase, went to my Postgres datasource, and clicked "Fetch Tables", to get my column
3. Updated a few rows with some UUIDs
4. In the design section, I added a `Data Provider` + `Table`
5. In `Define Filters` for the data provider, I set the filter:  `myUuid Equals 'a8bb354b-2791-4292-9da8-69f0332989cd'`

This filtered ok. So...

6. I added a form and an input above my table, and filtered based on the input
7. Opened my app and pasted a UUID into the search input.

Filtered correctly

Let me know if there's anything different in how you are doing it - happy to look into it more.

**Reply by wesshaw** [reply]
I appreciate you looking into it.  What I'm doing is basically the same thing, but I would notice an error notification where the uuid was truncated in the message.  I ended up just creating a custom query and that's working well... actually better because I can do some data transformation in the query.
